### PR TITLE
test: `arrow_path_to_parquet`

### PR DESCRIPTION
**Pull request by de-sh**
Fixes #XXXX.

### Description

Adds tests to validate working of the `arrow_path_to_parquet` function

<hr>

This PR has:
- [ ] been tested to ensure log ingestion and log query works.
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added documentation for new or modified features or behaviors.

## Summary by CodeRabbit

- **New Features**
  - Enhanced conversion from arrow file paths to parquet file paths, including improved error handling.
  - Added logging for unexpected filename structures to strengthen file handling.
- **Bug Fixes**
  - Updated file extension filtering logic to utilize a constant value for better maintainability.
- **Tests**
  - New test cases added to validate the file path conversion functionality across various scenarios and edge cases.

### CODE REVIEW

File: src/parseable/streams.rs [hunk 1]
```diff
@@ -67,6 +67,18 @@ use super::{
...
+fn arrow_path_to_parquet(path: &Path, random_string: &str) -> Option<PathBuf> {
+    let filename = path.file_stem()?.to_str()?;
+    let (_, front) = filename.split_once('.')?;
+    assert!(filename.contains('.'), "contains the delim `.`");
+    let filename_with_random_number = format!("{front}.data.{random_string}.parquet");
+    let mut parquet_path = path.to_owned();
+    parquet_path.set_file_name(filename_with_random_number);
+
+    Some(parquet_path)
+}
```
[the remark this answers is not in the document]
Comment (de-sh) [reply src/parseable/streams.rs]: Nice, thanks for the vigilance!

**Review by nikhilsinhaparseable — APPROVED** [review]
good to merge

File: src/parseable/streams.rs [hunk 2]
```diff
@@ -67,6 +67,18 @@ use super::{
...
+fn arrow_path_to_parquet(path: &Path, random_string: &str) -> Option<PathBuf> {
+    let filename = path.file_stem()?.to_str()?;
+    let (_, front) = filename.split_once('.')?;
+    assert!(front.contains('.'), "contains the delim `.`");
+    let filename_with_random_number = format!("{front}.{random_string}.parquet");
+    let mut parquet_path = path.to_owned();
+    parquet_path.set_file_name(filename_with_random_number);
+
+    Some(parquet_path)
+}
```
[the remark this answers is not in the document]
Comment (de-sh) [reply src/parseable/streams.rs:80]: front already carries ".data" at the end as asserted above
